### PR TITLE
Add timeout for cli/verify smoke test

### LINKED ISSUE #679: Add timeout to verify command
Currently, the `cypress verify` command can just hang because we do not put a timeout on a test run. It would be better to add a a timeout, and if the test really timed out, show the error output.

### LINKED ISSUE #819: Cypress hangs on verify step
- Operating System: CentOS 7.4
- Cypress Version: 1.0.2
- Browser Version: N/A

### Is this a Feature or Bug?
Bug

### Current behavior:
Cypress keeps "Verifying Cypress can run", so the program hangs

### Desired behavior:
Cypress to run normally

### How to reproduce:
```
npm install cypress mocha mocha-multi-reporters mocha-junit-reporter
node_modules/cypress/bin/cypress verify
```

### Additional Info (images, stack traces, etc)
No, are there log files somewhere?

## Cypress team update

If you hit problems running Cypress on a specific system:

- try using newer Cypress version
- try using our official images from [cypress-io/cypress-docker-images](https://github.com/cypress-io/cypress-docker-images)
- run Cypress with debug log output. In particular, we would like to see debug output from the CLI module and from our [XVFB module](https://github.com/cypress-io/xvfb). You can turn them both using `DEBUG=cypress:cli,xvfb npx cypress verify`
- provide an example for us to reproduce, and please remember that it will be hard for us to install an entire system that matches your system unless you provide Docker file
- see our examples for different CI systems that we run already in [cypress-example-kitchensink](https://github.com/cypress-io/cypress-example-kitchensink#ci-status)

### PR DESCRIPTION
**Pull request by kuceb**
fix #819 :smile: the smoke test should no longer hang after this
fix #679

this will users debug why the smoke test process could hang

we now
- kill smoketest after 10s using execa's timeout feature
- print an error message that the command timed out along with stderr (or stdout)

### Review tasks:
- [x] remove promise cancellation

### CODE REVIEW

File: cli/lib/util.js [hunk 1]
```diff
@@ -16,6 +16,10 @@ const pkg = require(path.join(__dirname, '..', 'package.json'))
 const logger = require('./logger')
 const debug = require('debug')('cypress:cli')
 
+Promise.config({
+  cancellation: true,
```
Comment (brian-mann) [inline cli/lib/util.js]: why is this necessary? I don't see us calling `promise.cancel()` anywhere?

**Comment by jennifer-shehane** [comment]
Hey @Bkucera - I would love to get this PR in as well, maybe you can take a look since it seems related. https://github.com/cypress-io/cypress/pull/3807

File: cli/lib/util.js (hunk 1 above)
Comment (Bkucera) [reply cli/lib/util.js]: that should be taken out, left in there when I tried using bluebird timeouts

**Comment by Bkucera** [comment]
@jennifer-shehane yes, I believe that PR should be closed in favor of this one
